mimic/rest/auth_api.py

```diff
@@ -151,17 +153,45 @@ def get_impersonation_token(self, request):
         except ValueError:
             request.setResponseCode(400)
             return json.dumps(invalid_resource("Invalid JSON request body"))
-
+        impersonator_token = request.getHeader("x-auth-token")
         expires_in = content['RAX-AUTH:impersonation']['expire-in-seconds']
         username = content['RAX-AUTH:impersonation']['user']['username']
-
+        impersonated_token = 'impersonated_token_' + text_type(uuid4())
         session = self.core.sessions.session_for_impersonation(username,
-                                                               expires_in)
+                                                               expires_in,
+                                                               impersonator_token,
+                                                               impersonated_token)
+
         return json.dumps({"access": {
-            "token": {"id": session.token,
+            "token": {"id": impersonated_token,
                       "expires": format_timestamp(session.expires)}
         }})
 
+    @app.route('/v2.0/tokens/<string:token_id>', methods=['GET'])
+    def validate_token(self, request, token_id):
+        """
+        Creates a new session for the given tenant_id and token_id
+        and always returns response code 200.
+        """
+        request.setResponseCode(200)
+        tenant_id = request.args.get('belongsTo')
+        if tenant_id:
+            tenant_id = tenant_id[0]
+        session = self.core.sessions.session_for_tenant_id(tenant_id, token_id)
+        response = get_token(
+            session.tenant_id,
+            response_token=session.token,
+            response_user_id=session.user_id,
+            response_user_name=session.username,
+        )
+        for each_session in session.impersonator_session_list:
```

Comment:
`session_impersonator_list` should be a straight-up dictionary, not a list of dictionaries; each `(impersonator, impersonatee)` tuple only gets a single token (so there should only ever be one entry in each one of the dictionaries in the list) and the token must be globally unique (not even just within the tenant, but within all of identity, because the lookup on the token starts at the top of identity with no other context).

Also, the lookup here ought to be a method on the session; `session.impersonator_session_for_token(impersonated_token_id)`?